### DISCUSSION TITLE
Scrollbar.init() not found

### DISCUSSION THREAD

**Discussion by radoslaw-redzik-ggsitc-com** [idiotWu/smooth-scrollbar#570] [Q&A]
Hi, I have such a problem.
I have loaded smooth-scrollbar.js file into my website. I have a div item with class="data-scrollbar". When javascript finds this element, it should run up a constructor. But it does not. I have an error in console: Uncaught TypeError: Cannot read properties of undefined (reading 'init')

It appears on this piece of code:
```
let Scrollbar
if (typeof Scrollbar !== typeof null) {
  if (document.querySelectorAll(".data-scrollbar").length) {
    Scrollbar = window.Scrollbar
    Scrollbar.init(document.querySelector('.data-scrollbar'), {
      continuousScrolling: false,
    })
  }
}
```

After `Scrollbar = window.Scrollbar` the variable `Scrollbar` is still undefined. AFAIK it should be `function e() {
                return null !== t && t.apply(this, arguments) || this
            }`.  And this should invoke this piece of code (**init**):
```
var q = function(t) {
            function e() {
                return null !== t && t.apply(this, arguments) || this
            }
            return function(t, e) {
                function i() {
                    this.constructor = t
                }
                n(t, e),
                t.prototype = null === e ? Object.create(e) : (i.prototype = e.prototype,
                new i)
            }(e, t),
            e.init = function(t, e) {
                if (!t || 1 !== t.nodeType)
                    throw new TypeError("expect element to be DOM Element, but got " + t);
                return U(),
                j.has(t) ? j.get(t) : new V(t,e)
            }
```
On the page there is much more js files. Is this possible there is a conflict between them? How to find, what js files causes conflict?
Can you help me? Why this is not working?
Regards
Radosław

**Comment by sadeghbarati** [comment]
Can you share your repository or live site?

**Comment by radoslaw-redzik-ggsitc-com** [comment]
Unfortunatelly this webpage is in restricted mode, not available publicly.

**Comment by sadeghbarati** [comment]
I don't exactly know what your project's structure is, is it React or Pure JS?

If you are using `import` (async) and you want to add `Scrollbar` function to `window` object 

```js
import Scrollbar from 'smooth-scrollbar`

window.Scrollbar = Scrollbar
```

make sure to use

```js
console.log(window.Scrollbar) // not working

window.addEventListener('load', () => {
  console.log(window.Scrollbar) // works
})
```
